Update from upstream (tag android-5.1.1_r2)

anatol opened the pull request.

anatol committed:
Update from upstream (tag android-5.1.1_r2)

pbatard commented:
I'm afraid I can't accept it. You reverted the changes I applied in b83f4b2 (open -> fopen, etc.) that ensured that the application could be used on Windows in a MinGW environment.

anatol committed:
Fix 'integer comparison' compile warning

mkbootimg/unmkbootimg.c: In function ‘load_file’:
mkbootimg/unmkbootimg.c:46:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if(fread(data, 1, sz, fd) != sz) goto oops;

anatol commented:
Thanks for the quick reply, your changes are restored.

It is a very good idea to push your changes upstream, thus you will not need to support it in your local repo. It probably a good idea to push unmkbootimg to upstream as well.

pbatard commented:
> It is a very good idea to push your changes upstream

I only do that when I'm planning to be actively involved with the upstream project, which I'm not planning to do here. From having been involved in multiple Open Source projects, I don't believe that one's duties stop at submitting a commit. Unfortunately, I don't see myself having the time to do any kind of follow up on this project right now, so I'd rather not submit it than create problems with the upstream project by not having them find that the submitter has gone M.I.A.

If someone asks for support about `unmkbootimg` after it's been integrated upstream, I'd like to be in a position to provide that help. But right now, my time is way to limited to commit on doing something like that, so I'll keep these changes private.

Besides, `unmkbootimg` is really something I created in 5 minutes, to work around [a problem I was having in 2013](http://pete.akeo.ie/2013/10/compiling-and-running-your-own-android.html). I have since moved on from trying to be involved in Android (again due to lack of time), so, apart from integrating patches that are kindly submitted to me (:smiley:), I'm not willing to perform any additional work on that project.

I have now reviewed your changes and will be pushing them shortly. Thanks again!
